app/assets/javascripts/discourse/routes/topic.js.es6

```diff
@@ -85,6 +85,10 @@ const TopicRoute = Discourse.Route.extend({
       this.controllerFor('modal').set('modalClass', 'history-modal');
     },
 
+    showGrantBadgeModal() {
+      showModal('grant-badge', { model: this.modelFor('topic'), title: 'admin.badges.grant_badge' });
```

Comment:
I recycled and used translations from `admin`, let me know if this is not okay.

Reply:
I'd say it's fine to reuse them. It's the same functionality.
The only place I'd probably use a new translation is the action in the wrench menu. There are size constraints and I imagine some language could have a need for using a shorter label...